middleware/authentication_test.go

```diff
@@ -0,0 +1,139 @@
+package middleware_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/lvl484/user-manager/middleware"
+	"github.com/lvl484/user-manager/mock"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/lvl484/user-manager/logger"
+	model "github.com/lvl484/user-manager/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	logger.SetLogger(&logger.LogConfig{Output: "Stdout", Level: "debug"})
+
+	code := m.Run()
+
+	os.Exit(code)
+}
+
+func TestBasicAuthenticationMiddlewareValidPass(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mock := mock.NewMockUserProvider(ctrl)
+
+	mock.EXPECT().GetInfo("i3odja").Return(returnedValue, nil)
+
+	ba := middleware.NewBasicAuthentication(mock)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
```

Comment:
Only one route registered is `GET /uuid`
Expected response code should be `404`, please check